demo/kserve/scripts/utils.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+
+die() {
+  color_red='\e[31m'
+  color_yellow='\e[33m'
+  color_reset='\e[0m'
+  printf "${color_red}FATAL:${color_yellow} $*${color_reset}\n" 1>&2
+  exit 10
+}
+
+info() {
+  color_blue='\e[34m'
+  color_reset='\e[0m'
+  printf "${color_blue}$*${color_reset}\n" 1>&2
+}
+
+success() {
+  color_green='\e[32m'
+  color_reset='\e[0m'
+  printf "${color_green}$*${color_reset}\n" 1>&2
+}
+
+check_pod_status() {
+  local -r JSONPATH="{range .items[*]}{'\n'}{@.metadata.name}:{@.status.phase}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}"
+  local -r pod_selector="$1"
+  local -r pod_namespace="$2"
+  local pod_status
+  local pod_entry
+
+  pod_status=$(oc get pods -l $pod_selector -n $pod_namespace -o jsonpath="$JSONPATH") 
```

Comment:
@Jooho, from what I've seen, this line is the one very verbose, I think it could be interesting to find a way to make it more friendly with `set -x`, so that the rest of the code can benefit from bash tracing.

On way to do it could be to use a [subshell](https://unix.stackexchange.com/questions/22796/can-i-export-functions-in-bash):
```
myfun() {
    echo "Hello!"
}

export -f myfun
./bash -c myfun
```
in your case, something like that:
```
_check_pod_status() {
...
}
export -f _check_pod_status
check_pod_status() {
  bash -ec check_pod_status
}
```